.changeset/two-pumpkins-wait.md

```diff
@@ -0,0 +1,49 @@
+---
+'@scale-codec/definition-compiler': major
+'@scale-codec/definition-runtime': major
+---
+
+**refactor**: use local opaque type pattern, remove `type-fest` dep (and `Opaque` re-export); workaround enum issues with `EnumBox` type; update generation, generate less code
```

Comment:
```suggestion
**refactor**: use local opaque type pattern, remove `type-fest` dep (and `Opaque` re-export); implement workaround enum issues with `EnumBox` type; update generation to produce less code
```

> pattern

is it just a single pattern?

> dep

dependency?

Reply:
> is it just a single pattern?

Yes

> dep

Yes

Reply:
Changed to

> use local opaque type pattern, remove `type-fest` dependency (and `Opaque` re-export accordingly); work around enum issues with `EnumBox` type; update generation to produce less code